model/pipeline/src/main/proto/beam_runner_api.proto

```diff
@@ -1158,6 +1158,90 @@ message SideInput {
   FunctionSpec window_mapping_fn = 3;
 }
 
+message StandardArtifacts {
+  enum Types {
+    // A URN for locally-accessible artifact files.
+    // payload: ArtifactFilePayload
+    FILE     = 0 [(beam_urn) = "beam:artifact:type:file:v1"];
+
+    // A URN for artifacts described by URLs.
+    // payload: ArtifactUrlPayload
+    URL      = 1 [(beam_urn) = "beam:artifact:type:url:v1"];
+
+    // A URN for artifacts embedded in ArtifactInformation proto.
+    // payload: EmbeddedFilePayload.
+    EMBEDDED = 2 [(beam_urn) = "beam:artifact:type:embedded:v1"];
+
+    // A URN for Python artifacts hosted on PYPI.
+    // payload: PypiPayload
+    PYPI     = 3 [(beam_urn) = "beam:artifact:type:pypi:v1"];
+
+    // A URN for Java artifacts hosted on a Maven repository.
+    // payload: MavenPayload
+    MAVEN    = 4 [(beam_urn) = "beam:artifact:type:maven:v1"];
+  }
+  enum Roles {
+    // A URN for staging-to role.
+    // payload: ArtifactStagingToRolePayload
+    STAGING_TO  = 0 [(beam_urn) = "beam:artifact:role:staging_to:v1"];
+  }
+}
+
+message ArtifactFilePayload {
+  // a string for an artifact path e.g. "/tmp/foo.jar"
+  string path = 1;
+
+  // The hex-encoded sha256 checksum of the artifact.
+  string sha256 = 2;
+}
+
+message ArtifactUrlPayload {
+  // a string for an artifact URL e.g. "https://.../foo.jar" or "gs://tmp/foo.jar"
+  string path = 1;
+}
+
+message EmbeddedFilePayload {
+  // raw data bytes for an embedded artifact
+  bytes data = 1;
```

Comment:
We could consider just letting the payload be the raw bytes (and similarly for other one-field payloads).

Reply:
Done.